subsys/bluetooth/host/hci_core.c

```diff
@@ -5464,13 +5445,80 @@ int bt_id_delete(u8_t id)
 	return 0;
 }
 
+#if defined(CONFIG_BT_PRIVACY)
+static void bt_read_identity_root(u8_t *ir)
+{
+#if defined(CONFIG_BT_HCI_VS_EXT)
+	struct bt_hci_rp_vs_read_key_hierarchy_roots *rp;
+	struct net_buf *rsp;
+	int err;
+#endif /* CONFIG_BT_HCI_VS_EXT */
+
+	/* Invalid IR */
+	memset(ir, 0, 16);
+
+#if defined(CONFIG_BT_HCI_VS_EXT)
+	if (!(bt_dev.vs_commands[1] & BIT(1))) {
+		return;
+	}
+
+	err = bt_hci_cmd_send_sync(BT_HCI_OP_VS_READ_KEY_HIERARCHY_ROOTS, NULL,
+		&rsp);
+	if (err) {
+		return;
+	}
+
+	rp = (void *)rsp->data;
+
+	BT_DBG("status 0x%02x", rp->status);
+	if (rp->status) {
```

Comment:
This is redundant since `rp->status` will always be zero if `bt_hci_send_cmd_sync()` returned zero